src/Microsoft.IdentityModel.Protocols/Configuration/ConfigurationManager.cs

```diff
@@ -199,7 +199,10 @@ public async Task<T> GetConfigurationAsync(CancellationToken cancel)
                     catch (Exception ex)
                     {
                         _syncAfter = DateTimeUtil.Add(now.UtcDateTime, _automaticRefreshInterval < _refreshInterval ? _automaticRefreshInterval : _refreshInterval);
-                        throw LogHelper.LogExceptionMessage(new InvalidOperationException(LogHelper.FormatInvariant(LogMessages.IDX20803, (_metadataAddress ?? "null")), ex));
+                        if (_currentConfiguration == null) // Throw an exception if there's no configuration to return.
+                            throw LogHelper.LogExceptionMessage(new InvalidOperationException(LogHelper.FormatInvariant(LogMessages.IDX20803, (_metadataAddress ?? "null")), ex));
+                        else
+                            LogHelper.LogWarning(LogHelper.FormatInvariant(LogMessages.IDX20806, (_metadataAddress ?? "null"), ex));
```

Comment:
Can we bump this up to error?
How about LogHelper.LogException with the same message, but don't throw?

Reply:
Suggestion: Can this only happen if there is a network connection issue or how can it happen that we don't have the configuration? If we have additional information from e.g. the http response we could possibly add that? (e.g. time-out vs. 500, vs. not found, etc.)? Perhaps also depending on the response we can decide if it's a warning or an error?

---
In reply to: [280398515](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/1157#discussion_r280398515) [](ancestors = 280398515)

Reply:
@henrik-me It depends on the retriever that was used. I have had conversations with users where they need details. For example, if an IdentityProvider accidentally pushes bad metadata {bad formatting for example}, that would cause an error.

Reply:
@brentschmaltz It looks like the HttpDocumentRetriever that's used by default in the OpenIdConnectConfigurationRetriever ends up throwing if the response does not have a success status code. So that exception could have some extra information as to why retrieving metadata failed.

I'm not sure how we would go about providing additional information about the HTTP response from the ConfigurationManager, as this information would somehow need to be bubbled up from the IDocumentRetriever that's used by the configuration retriever...

Reply:
If that information at that point in time is logged then perhaps the logs already have enough information for the developer, even though it may not be used to potentially improve this error message. I think that is ok, however I suggest reviewing what we log in the HttpDocumentRetriever.

Reply:
This discussion is closely related to https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/1112.
Maria is correct, `EnsureSuccessStatusCode `method throws `HttpRequestException` that doesn't contain an http status code.

I believe that there is value in bubbling-up an exception that has an http status code property on it, which allows developers to react to specific events (e.g. 429 for throttling).

Reply:
https://github.com/dotnet/corefx/issues/26477. Please don't invent a new exception type, we can address this in the framework instead.

Reply:
@Tratcher As noted above, HttpDocumentRetriever calls EnsureSuccessStatusCode(), which throws. I have had customers indicate there is additional info as stated in https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/1112. How do we serve that up?

Reply:
I removed the new exception type and modified this PR to also address #1112.